routing/endpointregistry.go

```diff
@@ -164,17 +166,17 @@ func (r *EndpointRegistry) updateStats() {
 			e.totalFailedRoundTrips[nextSlot].Store(0)
 			e.totalRequests[nextSlot].Store(0)
 
+			newDropProbability := 0.0
 			failed := e.totalFailedRoundTrips[curSlot].Load()
 			requests := e.totalRequests[curSlot].Load()
 			if requests > r.minRequests {
 				failedRoundTripsRatio := float64(failed) / float64(requests)
-				if failedRoundTripsRatio > 0.0 {
+				if failedRoundTripsRatio > r.minHealthCheckDropProbability {
 					log.Infof("Passive health check: marking %q as unhealthy due to failed round trips ratio: %0.2f", key, failedRoundTripsRatio)
+					newDropProbability = failedRoundTripsRatio
```

Comment:
Should we keep `min(failedRoundTripsRatio, r.maxHealthCheckDropProbability)` here?